app/src/main/java/org/mozilla/fenix/library/downloads/DownloadFragment.kt

```diff
@@ -73,12 +95,56 @@ class DownloadFragment : LibraryPageFragment<DownloadItem>(), UserInteractionHan
 
     override val selectedItems get() = downloadStore.state.mode.selectedItems
 
+    private fun invalidateOptionsMenu() {
+        activity?.invalidateOptionsMenu()
+    }
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
+        setHasOptionsMenu(true)
+    }
 
-        requireComponents.analytics.metrics.track(Event.HistoryOpened)
+    private fun displayDeleteAll() {
+        activity?.let { activity ->
+            AlertDialog.Builder(activity).apply {
+                setMessage(R.string.download_delete_all_dialog)
+                setNegativeButton(R.string.delete_browsing_data_prompt_cancel) { dialog: DialogInterface, _ ->
+                    dialog.cancel()
+                }
+                setPositiveButton(R.string.delete_browsing_data_prompt_allow) { dialog: DialogInterface, _ ->
+                    downloadStore.dispatch(DownloadFragmentAction.EnterDeletionMode)
+                    // Use fragment's lifecycle; the view may be gone by the time dialog is interacted with.
+                    lifecycleScope.launch(IO) {
+                        requireComponents.core.store.dispatch(DownloadAction.RemoveAllDownloadsAction)
```

Comment:
Why are we using the component store here but the `downloadStore.dispatch` above and below?